proof/invariant-abstract/IpcDet_AI.thy

```diff
@@ -123,11 +399,22 @@ lemma tcb_ep_append_valid_ep:
    \<lbrace>\<lambda>q'. valid_ep (update_ep_queue ep q')\<rbrace>"
   by (cases ep) (wpsimp wp: tcb_ep_append_valid_SendEP tcb_ep_append_valid_RecvEP)+
 
+lemma set_map_fst_filter_zip:
+  "set (map fst (filter P (zip xs ys))) \<subseteq> set xs"
```

Comment:
another lib candidate; if you're ever in the mood for janitorial work, there's a fair few things I've seen in this review we could move to lib